crates/forge-runner/src/lib.rs

```diff
@@ -292,19 +292,75 @@ fn choose_test_strategy_and_run(
     runner: Arc<SierraCasmRunner>,
     runner_config: Arc<RunnerConfig>,
     runner_params: Arc<RunnerParams>,
-    send: &Sender<()>,
-) -> JoinHandle<Result<TestCaseSummary>> {
+    send: Sender<()>,
+) -> JoinHandle<Result<AnyTestCaseSummary>> {
     if args.is_empty() {
-        run_test(case, runner, runner_config, runner_params, send.clone())
+        tokio::task::spawn(async move {
+            let res = run_test(case, runner, runner_config, runner_params, send).await??;
+            Ok(AnyTestCaseSummary::Single(res))
+        })
     } else {
-        run_with_fuzzing(
-            args,
-            case,
-            runner,
-            runner_config,
-            runner_params,
-            send.clone(),
-        )
+        tokio::task::spawn(async move {
+            let res =
+                run_with_fuzzing(args, case, runner, runner_config, runner_params, send).await??;
+            Ok(AnyTestCaseSummary::Fuzzing(res))
+        })
+    }
+}
+
+fn fuzzing_run_summary(results: Vec<TestCaseSummary<Single>>) -> TestCaseSummary<Fuzzing> {
+    let last: TestCaseSummary<Single> = results
+        .iter()
+        .last()
+        .cloned()
+        .expect("Fuzz test should always run at least once");
+    match last {
```

Comment:
That's kind of weird, we're matching on the last item in the test set and if it has succeeded, we automatically assume all of them did?

Reply:
Yes, I started the refactor to hide this logic because it is not intuitive. If fuzzing fails, then we cancel the whole run so only the last result matters.

Reply:
I will add a comment about this

Reply:
Done